parsl/executors/globus_compute.py

```diff
@@ -0,0 +1,139 @@
+import uuid
+from concurrent.futures import Future
+from typing import Any, Callable, Dict, Optional, Union
+
+from parsl.errors import OptionalModuleMissing
+from parsl.executors.base import ParslExecutor
+from parsl.utils import RepresentationMixin
+
+UUID_LIKE_T = Union[uuid.UUID, str]
+
+
+class GlobusComputeExecutor(ParslExecutor, RepresentationMixin):
+    """ GlobusComputeExecutor enables remote execution on Globus Compute endpoints
+
+    GlobusComputeExecutor is a thin wrapper over globus_compute_sdk.Executor
+    Refer to `globus-compute user documentation <https://globus-compute.readthedocs.io/en/latest/executor.html>`_
+    and `reference documentation <https://globus-compute.readthedocs.io/en/latest/reference/executor.html>`_
+    for more details.
+    """
+
+    def __init__(
+            self,
+            endpoint_id: Optional[UUID_LIKE_T] = None,
+            task_group_id: Optional[UUID_LIKE_T] = None,
+            resource_specification: Optional[dict[str, Any]] = None,
+            user_endpoint_config: Optional[dict[str, Any]] = None,
+            label: str = "GlobusComputeExecutor",
+            batch_size: int = 128,
+            amqp_port: Optional[int] = None,
+            **kwargs,
+    ):
```

Comment:
Consider adding the `client` parameter

Reply:
I'm hesitant to add this because it will require `globus_compute_sdk.sdk.Client` type hints when this is an Optional module. Right now, passing this in via **kwargs will still work fine. What do you think?

Reply:
I think we should add it to clearly communicate support for the parameter. I've already had an early adopter ask about this.

To add type hints, we could do something like this:
```python
try:
    from globus_compute_sdk import Client
except ImportError:
    Client = Any
 
 class GlobusComputeExecutor(ParslExecutor, RepresentationMixin):
    def __init__(
            self,
            endpoint_id: Optional[UUID_LIKE_T] = None,
            client: Optional[Client] = None,
            ...
```

or

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from globus_compute_sdk import Client

 class GlobusComputeExecutor(ParslExecutor, RepresentationMixin):
    def __init__(
            self,
            endpoint_id: Optional[UUID_LIKE_T] = None,
            client: Optional["Client"] = None,
            ...
```

Reply:
haha thanks Ben :metal:

Reply:
Fixed now.

Reply:
I could be missing something, but I don't see the `client` parameter.

Reply:
It's here -> https://github.com/Parsl/parsl/pull/3607/files#diff-03350429687012cb1a3ed39ac9d3549f2b4c825555f7e9dee0600ff82b16164aR48